applications/ShallowWaterApplication/python_scripts/coupling/depth_integration_input_process.py

```diff
@@ -173,15 +182,15 @@ def _CreateMapper(self):
                 "search_radius" : 0.0
             }
         }""")
-        min_point = KM.Point([ 1e6,  1e6,  1e6])
-        max_point = KM.Point([-1e6, -1e6, -1e6])
-        for node in self.interface_model_part.Nodes:
-            for i in range(3):
-                point = KM.Point(node)
-                min_point[i] = min([min_point[i], point[i]])
-                max_point[i] = max([max_point[i], point[i]])
-        distance = 1.05 * (max_point - min_point).norm_2()
-        mapper_settings["search_settings"]["search_radius"].SetDouble(distance)
+        domain_size = self.input_model_part.ProcessInfo[KM.DOMAIN_SIZE]
+        if domain_size == 2:
+            min_point, max_point = KM.BoundingBox(self.interface_model_part.Nodes).GetPoints()
+            search_radius = 1.05 * (max_point - min_point).norm_2()
+        elif domain_size == 3:
+            search_radius = self.interface_model_part.Conditions.__iter__().__next__().GetGeometry().Length()
+        else:
+            raise Exception("DepthIntegrationInputProcess._CreateMapper: The domain size of the input_model_part is: ", domain_size)
+        mapper_settings["search_settings"]["search_radius"].SetDouble(search_radius)
```

Comment:
is it slow without specifying this?
I recently updated the search and made it a lot faster, but considering that you are using the new input I guess you are aware of it

Reply:
Thanks for the comments @philbucher 
In some cases I have a cloud of points. Since accuracy is not crucial, its better to use the nearest neighbor in all the cases.
For the 3D case, now I'm using the default search radius, its cleaner